lib/checks.js

```diff
@@ -1,26 +1,112 @@
+  "no-cond-assign": "Bug Risk",
   "no-console": "Bug Risk",
   "no-constant-condition": "Bug Risk",
+  "no-control-regex": "Bug Risk",
   "no-debugger": "Bug Risk",
   "no-delete-var": "Bug Risk",
+  "no-div-regex": "Bug Risk",
+  "no-dupe-args": "Bug Risk",
   "no-dupe-keys": "Bug Risk",
   "no-duplicate-case": "Bug Risk",
+  "no-else-return": "Clarity",
+  "no-empty": "Bug Risk",
   "no-empty-character-class": "Bug Risk",
+  "no-empty-label": "Bug Risk",
+  "no-empty-pattern": "Bug Risk",
+  "no-eq-null": "Bug Risk",
+  "no-eval": "Security",
+  "no-ex-assign": "Bug Risk",
+  "no-extend-native": "Bug Risk",
+  "no-extra-bind": "Bug Risk",
+  "no-extra-boolean-cast": "Bug Risk",
+  "no-extra-parens": "Bug Risk",
+  "no-extra-semi": "Bug Risk",
   "no-fallthrough": "Bug Risk",
+  "no-floating-decimal": "Clarity",
   "no-func-assign": "Bug Risk",
+  "no-implicit-coercion": "Bug Risk",
+  "no-implied-eval": "Security",
   "no-inner-declarations": "Compatibility",
   "no-invalid-regexp": "Bug Risk",
+  "no-invalid-this": "Bug Risk",
   "no-irregular-whitespace": "Compatibility",
+  "no-iterator": "Compatibility",
+  "no-label-var": "Bug Risk",
+  "no-labels": "Bug Risk",
+  "no-lone-blocks": "Bug Risk",
+  "no-loop-func": "Bug Risk",
+  "no-magic-numbers": "Clarity",
+  "no-mixed-requires": "Clarity",
+  "no-multi-spaces": "Bug Risk",
+  "no-multi-str": "Compatibility",
+  "no-native-reassign": "Bug Risk",
   "no-negated-in-lhs": "Bug Risk",
+  "no-new": "Bug Risk",
+  "no-new-func": "Clarity",
+  "no-new-require": "Clarity",
+  "no-new-wrappers": "Bug Risk",
   "no-obj-calls": "Bug Risk",
   "no-octal": "Compatibility",
+  "no-octal-escape": "Compatibility",
+  "no-param-reassign": "Bug Risk",
+  "no-path-concat": "Bug Risk",
+  "no-process-env": "Bug Risk",
+  "no-process-exit": "Bug Risk",
+  "no-proto": "Compatibility",
+  "no-redeclare": "Bug Risk",
+  "no-regex-spaces": "Bug Risk",
+  "no-restricted-modules": "Security",
+  "no-return-assign": "Bug Risk",
+  "no-script-url": "Security",
+  "no-self-compare": "Bug Risk",
+  "no-sequences": "Bug Risk",
+  "no-shadow": "Bug Risk",
+  "no-shadow-restricted-names": "Bug Risk",
+  "no-sparse-arrays": "Bug Risk",
+  "no-sync": "Bug Risk",
+  "no-throw-literal": "Clarity",
   "no-undef": "Bug Risk",
+  "no-undef-init": "Bug Risk",
+  "no-undefined": "Compatibility",
+  "no-unexpected-multiline": "Bug Risk",
   "no-unreachable": "Bug Risk",
+  "no-unused-expressions": "Bug Risk",
+  "no-unused-vars": "Bug Risk",
+  "no-use-before-define": "Compatibility",
+  "no-useless-call": "Bug Risk",
+  "no-useless-concat": "Bug Risk",
+  "no-void": "Compatibility",
+  "no-warning-comments": "Bug Risk",
+  "no-with": "Compatibility",
+  "np-dupe-keys": "Bug Risk",
+  "radix": "Clarity",
   "use-isnan": "Bug Risk",
-  "valid-typeof": "Bug Risk"
+  "use-strict": "Bug Risk",
+  "valid-jsdoc": "Clarity",
+  "valid-typeof": "Bug Risk",
+  "vars-on-top": "Clarity",
+  "wrap-iife": "Clarity",
+  "yoda": "Clarity"
```

Comment:
We are not following each other very well. :)

I understand what Yoda conditionals are: I personally tend to use them. The reason I, personally, use Yoda conditionals is to *avoid* the bug risk of accidental `=`. So I would consider *not* using a Yoda conditional a bug risk, not a clarity decision. That is:

`if (color == "red") { ... }`

is, in my eyes, a bug risk (because it's far too easy to miss/delete one of the `=` and not realize it). (The risk is somewhat reduced by the fact that almost everybody requires `===` in JS nowadays, but still.)

That being said, "Bug Risk" may not be an appropriate classification because it only makes sense if your eslintrc requires people *to use* yoda conditionals. If your eslintrc requires them *not* to be used, "Clarity" does make more sense. So since either configuration is possible, perhaps "Clarity" is more broadly applicable.